[2024-07-19T21:35:41.302+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-07-19T21:35:41.324+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_reddit_pipeline.reddit_extraction manual__2024-07-19T21:35:40.435179+00:00 [queued]>
[2024-07-19T21:35:41.330+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_reddit_pipeline.reddit_extraction manual__2024-07-19T21:35:40.435179+00:00 [queued]>
[2024-07-19T21:35:41.331+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2024-07-19T21:35:41.344+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): reddit_extraction> on 2024-07-19 21:35:40.435179+00:00
[2024-07-19T21:35:41.351+0000] {standard_task_runner.py:64} INFO - Started process 78 to run task
[2024-07-19T21:35:41.353+0000] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'etl_reddit_pipeline', 'reddit_extraction', 'manual__2024-07-19T21:35:40.435179+00:00', '--job-id', '15', '--raw', '--subdir', 'DAGS_FOLDER/reddit_dag.py', '--cfg-path', '/tmp/tmpozsenr6d']
[2024-07-19T21:35:41.355+0000] {standard_task_runner.py:91} INFO - Job 15: Subtask reddit_extraction
[2024-07-19T21:35:41.393+0000] {task_command.py:426} INFO - Running <TaskInstance: etl_reddit_pipeline.reddit_extraction manual__2024-07-19T21:35:40.435179+00:00 [running]> on host aa0946d1cd0d
[2024-07-19T21:35:41.460+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='nopega' AIRFLOW_CTX_DAG_ID='etl_reddit_pipeline' AIRFLOW_CTX_TASK_ID='reddit_extraction' AIRFLOW_CTX_EXECUTION_DATE='2024-07-19T21:35:40.435179+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-07-19T21:35:40.435179+00:00'
[2024-07-19T21:35:41.461+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-07-19T21:35:41.477+0000] {logging_mixin.py:188} INFO - connect to reddit!
[2024-07-19T21:35:42.791+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-07-19T21:35:42.791+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/pipelines/reddit_pipeline.py", line 13, in reddit_pipeline
    load_data_to_csv(post_df, OUTPUT_PATH)
  File "/opt/airflow/etls/reddit_etl.py", line 39, in load_data_to_csv
    data.to_csv(path,index=False)
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/generic.py", line 3551, in to_csv
    return DataFrameRenderer(formatter).to_csv(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/formats/format.py", line 1180, in to_csv
    csv_formatter.save()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/formats/csvs.py", line 241, in save
    with get_handle(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/io/common.py", line 789, in get_handle
    handle = open(
IsADirectoryError: [Errno 21] Is a directory: '/opt/airflow/data/output'
[2024-07-19T21:35:42.807+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=etl_reddit_pipeline, task_id=reddit_extraction, run_id=manual__2024-07-19T21:35:40.435179+00:00, execution_date=20240719T213540, start_date=20240719T213541, end_date=20240719T213542
[2024-07-19T21:35:42.816+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 15 for task reddit_extraction ([Errno 21] Is a directory: '/opt/airflow/data/output'; 78)
[2024-07-19T21:35:42.848+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2024-07-19T21:35:42.861+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-07-19T21:35:42.863+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
